File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/mcp_server/tool_registry.py
from tools.websearch import web_search


class ToolRegistry:
    """Simple registry that maps tool names to callable functions."""

    def __init__(self):
        self.tools: dict = {}
        self._register_default_tools()

    def _register_default_tools(self):
        self.register("web_search", web_search)

    def register(self, name: str, func):
        self.tools[name] = func

    def get(self, name: str):
        if name not in self.tools:
            raise KeyError(f'Tool "{name}" not found.')
        return self.tools[name]

File: MCP_GATEWAY/SKill_SOCKET-Mcp_gateway/mcp_server/toolRegistry.js
const { webSearch } = require('../tools/websearch');
class ToolRegistry{
    constructor(){
        this.tools={};
        this.registerDefaultTools();
    }
    registerDefaultTools(){
         this.register('web_search', webSearch);
    }
    register(name,func){
        this.tools[name]=func;
    }
    get(name){
         if (!this.tools[name]) throw new Error(`Tool "${name}" not found.`);
        return this.tools[name];
    }
}
module.exports=ToolRegistry